### PR TITLE
Switch to modern Travis CI infrastructure

### LINKED ISSUE #310: Switch to modern Travis CI infrastructure
We are currently using the old, slow, Travis CI infrastructure.

We need to do the following to get on the new one:

- Explicitly list packages we depend upon from apt. It doesn't look like there is an option for doing build dependencies easily like we do now. See: http://docs.travis-ci.com/user/migrating-from-legacy/#Adding-APT-Packages
- Use `--prefix=$HOME/irssi-build` (since we don't have `sudo` we cannot do `make install` to `/`).
- Add `sudo: false` in `.travis.yml`.  See: http://docs.travis-ci.com/user/migrating-from-legacy/

While you are at it, please enable `--enable-silent-rules` for the call to `autogen.sh` in `.travis.yml`. This should make warnings stand out more clearly.

### PR DESCRIPTION
**Pull request by Jellyfrog**
Had to switch from `lynx` to `elinks` due to; https://github.com/travis-ci/apt-package-whitelist/issues/630

Fixes #310

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,19 +8,27 @@ env:
     - CC=clang
     - CC=gcc
 
+addons:
+    apt:
+        packages:
+            - libc6
+            - libglib2.0-0
+            - libssl1.0.0
+            - libtinfo5
```
Comment (dequis) [inline .travis.yml]: Pretty sure these first four deps can be nuked right away.

Comment (Jellyfrog) [inline .travis.yml]: I simply copied the list from ubuntu's deps.
I'll update and we see.

**Comment by dequis** [comment]
>checking for working Perl support... error linking with perl libraries, building without Perl

In the system-perl ones.

https://travis-ci.org/irssi/irssi/jobs/82764669#L451

>Error linking with perl libraries: -Wl,-E -fstack-protector -L/usr/local/lib -L/usr/lib/perl/5.14/CORE -lperl -ldl -lm -lpthread -lcrypt: /usr/bin/ld: cannot find -lperl
>
>clang: error: linker command failed with exit code 1 (use -v to see invocation)
>
>Try: sudo apt-get install libperl-dev

**Comment by Jellyfrog** [comment]
Fixed now.
Maybe enable true color and DANE?

**Comment by dequis** [comment]
Hmm, no benefit in enabling that here IMO, I'm just going to merge this as-is.

Thanks for your contribution!

**Pull request merged**